pkg/user/tx_client_test.go

```diff
@@ -281,3 +313,35 @@ func (suite *TxClientTestSuite) queryCurrentBalance(t *testing.T) int64 {
 	require.NoError(t, err)
 	return balanceResp.Balances.AmountOf(app.BondDenom).Int64()
 }
+
+func wasRemovedFromTxTracker(txHash string, txClient *user.TxClient) bool {
+	nonce, signer, exists := txClient.GetTxFromTxTracker(txHash)
+	return !exists && nonce == 0 && signer == ""
+}
+
+// asserts that a tx was indexed in the tx tracker and that the sequence does not increase
+func assertTxInTxTracker(t *testing.T, txClient *user.TxClient, txHash string, expectedSigner string) {
+	nonce, signer, exists := txClient.GetTxFromTxTracker(txHash)
+	require.True(t, exists)
+	require.Equal(t, expectedSigner, signer)
+	seq := txClient.Signer().Account(expectedSigner).Sequence()
+	// Successfully broadcast transaction increases the nonce
+	// txInfo is indexed before the nonce is increased
+	require.Equal(t, seq, nonce+1)
+}
+
+func setupTxClient(t *testing.T, ttlDuration time.Duration) (encoding.Config, *user.TxClient, testnode.Context) {
+	encCfg := encoding.MakeConfig(app.ModuleEncodingRegisters...)
+	defaultTmConfig := testnode.DefaultTendermintConfig()
+	defaultTmConfig.Mempool.TTLDuration = ttlDuration
+	testnodeConfig := testnode.DefaultConfig().
+		WithTendermintConfig(defaultTmConfig).
+		WithFundedAccounts("a", "b", "c").
+		WithAppCreator(testnode.CustomAppCreator("0utia"))
+	ctx, _, _ := testnode.NewNetwork(t, testnodeConfig)
+	_, err := ctx.WaitForHeight(1)
+	require.NoError(t, err)
+	txClient, err := user.SetupTxClient(ctx.GoContext(), ctx.Keyring, ctx.GRPCClient, encCfg, user.WithGasMultiplier(1.2))
+	require.NoError(t, err)
+	return encCfg, txClient, ctx
+}
```

Comment:
nit: this doesn't make too much sense as it's own function because it's so specific to one test.